compiler/shared_ast/print.ml

```diff
@@ -114,8 +114,11 @@ let rec typ (ctx : decl_ctx option) (fmt : Format.formatter) (ty : typ) : unit =
         punctuation "]")
   | TOption t -> Format.fprintf fmt "@[<hov 2>%a@ %a@]" base_type "option" typ t
   | TArrow (t1, t2) ->
-    Format.fprintf fmt "@[<hov 2>%a %a@ %a@]" typ_with_parens t1 op_style "→"
-      typ t2
+    Format.fprintf fmt "@[<hov 2>%a %a@ %a@]"
+      (Format.pp_print_list
+         ~pp_sep:(fun fmt () -> Format.fprintf fmt "@ %a@ " op_style "→")
+         typ_with_parens)
+      t1 op_style "→" typ t2
```

Comment:
I aggree with your proposal. with a tiny modification: if there is only one argument, we should not print parenthesis right ?

```
  | TArrow ([t1], t2) ->
    Format.fprintf fmt "@[<hov 2>%a@ →@ %a@]" format_typ_with_parens t1
      format_typ t2
  | TArrow (t1, t2) ->
    Format.fprintf fmt "@[<hov 2>(%a)@ →@ %a@]"
      (Format.pp_print_list
         ~pp_sep:(fun fmt () -> Format.fprintf fmt ",@ ")
         format_typ_with_parens)
      t1 format_typ t2
```

Reply:
Indeed!

Reply:
Hm that could be an ambiguity for functions taking a tuple as argument though ; if we have them at some point ? (but maybe it's the syntax for tuples that could be changed ?)

At the moment the (temporary) syntax for defining functions has commas but no parens around the argument list, so `(arg1, arg2, ... argn → ret)` is another possibility consistent with that (but probably less readable ? I'm unsure)

Anyway don't hold the PR for this it can always be changed later if we decide we have a better solution.